### ISSUE TITLE
Weird version semantics

### ISSUE THREAD

**Issue by PVince81** [owncloud/ocis-pkg#53]
In other modules like ocis-settings, we are importing the so-called v2 of ocis-pkg as `github.com/owncloud/ocis-pkg/v2 v2.2.2-0.20200812103920-db41b5a3d14d` (when using go get with `@master`)

However the commit "db41b5a3d14d" is on master, and not on the "v2" branch.
And the "v2" branch seems much older than master.

Is this intended ? What is the purpose of the "v2" branch then ?
And where is the old v1 version then ?

@butonic @refs

**Comment by refs** [comment]
`v2` should act as master. As per go modules docs:

> The recommended strategy is to develop v2+ modules in a directory named after the major version > suffix.

>github.com/googleapis/gax-go @ master branch
>/go.mod    → module github.com/googleapis/gax-go
>/v2/go.mod → module github.com/googleapis/gax-go/v2


However we're not versioning on a version folder, instead on a branch. When that's the case:

> Other strategies may keep major versions on separate branches. However, if v2+ source code is on the repository's default branch (usually master), tools that are not version-aware — including the go command in GOPATH mode — may not distinguish between major versions.

To my understanding we have to rebase master onto v2.

**Comment by PVince81** [comment]
yeah, because recently we updated using `@master` so I'm worried that we might have the wrong code.

however I saw that @butonic has pushed to ocis-pkg master, so probably that one is still the one we consider "the right one"

**Comment by refs** [comment]
@PVince81 https://github.com/owncloud/ocis-pkg/pull/54. For the future, after all this release party.
